src/filesystem/interface.js

```diff
@@ -206,6 +206,48 @@ function FileSystem(options, callback) {
     return watcher;
   };
 
+  //Object that uses filenames as keys
+  const statWatchers = new Map(); 
+
+  this.watchFile = function(filename, options, listener) {
+    const prevStat, currStat; 
+
+    if (Path.isNull(filename)) {
+      throw new Error('Path must be a string without null bytes.');
+    }
+    //Checks to see if there were options passed in and if not, the callback function will be set here
+    if (typeof options === 'function') {
+      listener = options; 
+      options = {};
+    }
+    //default 5007ms interval, persistent is not used this project
+    const interval = options.interval || 5007;
+    listener = listener || nop;
+    
+    //Stores initial prev value to compare 
+    fs.stat(filename, function(err, stats) { prevStat = stats}); 
+
+    //stores interval return values
+    statWatchers.set(filename, value);
+
+    var value = setInterval(function() {
+      fs.stat(filename, function(err, stats) {
+        if(err) {
+          console.log(err);
```

Comment:
![image](https://user-images.githubusercontent.com/37606829/55122332-eb26fb00-50d4-11e9-834e-b2a33b13a549.png)

Reply:
Any errors handling in nodejs seems to be happening in the highlighted area
![image](https://user-images.githubusercontent.com/37606829/55122389-2f1a0000-50d5-11e9-8710-50f27e5fc829.png)
I tried to analyze the start method, but it seems to be too complex for me with the terminology being used here, but I believe the highlighted portion below is what's important. 
![image](https://user-images.githubusercontent.com/37606829/55122503-9fc11c80-50d5-11e9-8446-04fc08dad0b7.png)

Reply:
Node is able to throw the error here because they are doing the initial `stat` synchronously, so the ultimate recipient of the `throw` would be the caller of `fs.watchFile`.  We can only do asynchronous calls in Filer and the Browser (i.e., we can't block on access to IndexedDB), so throwing doesn't make as much sense for us.

Another problem I see is that node is returning the `stat` object it gets back from that synchronous call.  We also can't do this.

I'm not 100% sure what the right approach is here.  Perhaps we need to kill the `interval` and have the watcher just become a no-op if there is an error?  Maybe logging an error to the console in addition does make sense.  Perhaps you could do:

`console.warn(`[Filer Error] fs.watchFile encountered an error with ${path}: ${err.message}`)`